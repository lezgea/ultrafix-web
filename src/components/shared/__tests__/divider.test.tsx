import React from 'react';
import { render } from '@testing-library/react';
import Divider from '../divider';


describe('Divider Component', () => {
    it('should render with default props', () => {
        const { container, asFragment } = render(<Divider />);
        const divider = container.firstChild;

        expect(divider).toHaveClass('w-full my-2 border-t');
        expect(divider).toHaveStyle({ borderColor: 'gray-100', borderTopWidth: '1px' });
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render with custom color', () => {
        const { container, asFragment } = render(<Divider color="red" />);
        const divider = container.firstChild;

        expect(divider).toHaveStyle({ borderColor: 'red' });
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render with custom thickness', () => {
        const { container, asFragment } = render(<Divider thickness="2px" />);
        const divider = container.firstChild;

        expect(divider).toHaveStyle({ borderTopWidth: '2px' });
        expect(asFragment()).toMatchSnapshot();
    });

    it('should render with custom marginY', () => {
        const { container, asFragment } = render(<Divider marginY="my-4" />);
        const divider = container.firstChild;

        expect(divider).toHaveClass('w-full my-4 border-t');
        expect(asFragment()).toMatchSnapshot();
    });

    it('should apply all custom props', () => {
        const { container, asFragment } = render(
            <Divider color="blue" thickness="3px" marginY="my-6" />
        );
        const divider = container.firstChild;

        expect(divider).toHaveClass('w-full my-6 border-t');
        expect(divider).toHaveStyle({ borderColor: 'blue', borderTopWidth: '3px' });
        expect(asFragment()).toMatchSnapshot();
    });
});
