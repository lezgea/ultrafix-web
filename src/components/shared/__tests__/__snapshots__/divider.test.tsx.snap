// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should apply all custom props 1`] = `
<DocumentFragment>
  <div
    class="w-full my-6 border-t"
    style="border-color: blue; border-top-width: 3px;"
  />
</DocumentFragment>
`;

exports[`Divider Component should render with custom color 1`] = `
<DocumentFragment>
  <div
    class="w-full my-2 border-t"
    style="border-color: red; border-top-width: 1px;"
  />
</DocumentFragment>
`;

exports[`Divider Component should render with custom marginY 1`] = `
<DocumentFragment>
  <div
    class="w-full my-4 border-t"
    style="border-top-width: 1px;"
  />
</DocumentFragment>
`;

exports[`Divider Component should render with custom thickness 1`] = `
<DocumentFragment>
  <div
    class="w-full my-2 border-t"
    style="border-top-width: 2px;"
  />
</DocumentFragment>
`;

exports[`Divider Component should render with default props 1`] = `
<DocumentFragment>
  <div
    class="w-full my-2 border-t"
    style="border-top-width: 1px;"
  />
</DocumentFragment>
`;
